langchain/src/chat_models/baiduwenxin.ts

```diff
@@ -0,0 +1,459 @@
+                  result: message.result,
+                  need_clear_history: message.need_clear_history,
+                  usage: message.usage,
+                };
+              } else {
+                response.result += message.result;
+                response.created = message.created;
+                response.need_clear_history = message.need_clear_history;
+                response.usage = message.usage;
+              }
+
+              // TODO this should pass part.index to the callback
+              // when that's supported there
+              // eslint-disable-next-line no-void
+              void runManager?.handleLLMNewToken(message.result ?? "");
+
+              if (message.is_end) {
+                if (resolved || rejected) {
+                  return;
+                }
+                resolved = true;
+                resolve(response);
+              }
+            }
+          ).catch((error) => {
+            if (!rejected) {
+              rejected = true;
+              reject(error);
+            }
+          });
+        })
+      : await this.completionWithRetry(
+          {
+            ...params,
+            messages: messagesMapped,
+          },
+          false,
+          options?.signal
+        );
+
+    const {
+      completion_tokens: completionTokens,
+      prompt_tokens: promptTokens,
+      total_tokens: totalTokens,
+    } = data.usage ?? {};
+
+    if (completionTokens) {
+      tokenUsage.completionTokens =
+        (tokenUsage.completionTokens ?? 0) + completionTokens;
+    }
+
+    if (promptTokens) {
+      tokenUsage.promptTokens = (tokenUsage.promptTokens ?? 0) + promptTokens;
+    }
+
+    if (totalTokens) {
+      tokenUsage.totalTokens = (tokenUsage.totalTokens ?? 0) + totalTokens;
+    }
+
+    const generations: ChatGeneration[] = [];
+    const text = data.result ?? "";
+    generations.push({
+      text,
+      message: new AIChatMessage(text),
+    });
+    return {
+      generations,
+      llmOutput: { tokenUsage },
+    };
+  }
+
+  /** @ignore */
+  async completionWithRetry(
+    request: ChatCompletionRequest,
+    stream: boolean,
+    signal?: AbortSignal,
+    onmessage?: (event: MessageEvent) => void
+  ) {
+    // The first run will get the accessToken
+    if (!this.accessToken) {
+      this.accessToken = await this.getAccessToken();
+    }
+
+    const makeCompletionRequest = async () => {
+      const url = `${this.apiUrl}?access_token=${this.accessToken}`;
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(request),
+        signal: signal,
+      });
+
+      if (!stream) {
+        return response.json();
+      } else {
+        if (response.body) {
+          const reader = response.body.getReader();
+
+          const decoder = new TextDecoder("utf-8");
+          let data = "";
+
+          while (true) {
```

Comment:
Constant condition here breaks our linter - can you try using `while (!done)` instead?

Reply:
done